### ISSUE TITLE
Is it safe to enter credit card information when using Cloudflare's flexible SSL?

### ISSUE THREAD

**Issue by NinoSkopac** [braintree/braintree-web-drop-in#154]
Hey

Using CF's flexible SSL means that the connection [is not encrypted](https://webmasters.stackexchange.com/questions/67930/are-there-any-disadvantages-to-cloudflare-s-flexible-ssl) between CF and origin server. When using your SDK, are CC numbers visible during that route? I've used Stripe in the past and I know they use an iframe to make it secure.

What does Braintree do?

**Comment by crookedneighbor** [comment]
Drop-in uses the `braintree-web` Hosted Fields component, which puts the credit card information (number, expiration date, cvv and postal code) inside iframes. It is equivalent to Stripe in that regard.

Since the credit card data is never on your site (it's all contained within the iframes), all you ever send to your server is the `nonce`, which has no use without the braintree server sdk initialized with _your_ credentials.

However, there is nothing preventing a man in the middle attacker from modifying the response between your server and cloudflare. An attacker could rewrite your page, and replace the drop-in with their own form that sends the credit card information to their server.

Cloudfare's offering is primarily intended for sites like personal blogs, where no sensitive information is sent to your server. (See https://support.cloudflare.com/hc/en-us/articles/200170416-What-do-the-SSL-options-mean-)

So, in short, even though Braintree (and Stripe) use iframes to capture the data, they should not be used with Cloudflare's Flexible SSL offering.

**Comment by NinoSkopac** [comment]
Cool, thanks for the explanation.
